### PR TITLE
Migrated concreate descriptor to be object based

### PR DESCRIPTION
**Pull request by dbecvarik**
Now every important part of the descriptor is treated as object. With
this we can easily extend our schema and use object api instead of
plain dictionaries inside code.

This also enables to have multiple schemas per object type, so we can
easily introduce incompatible changes to our schema without breaking
old images. Each object is resposinble to handling its schema and
converting it to compatible object state.

### CODE REVIEW

File: concreate/generator.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class Generator(object):
 
     def __init__(self, descriptor_path, target, overrides):
 
-        self.descriptor = Descriptor(descriptor_path, 'image').process()
+        self.descriptor = Image(tools.load_descriptor(descriptor_path))
```
Comment (goldmann) [inline concreate/generator.py]: I would say `self.descriptor` is a misleading name for an field now.

File: concreate/generator.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def prepare_artifacts(self):
 
     def override(self, overrides_path):
         logger.info("Using overrides file from '%s'." % overrides_path)
-        descriptor = Descriptor(overrides_path, 'overrides').process()
+        descriptor = Overrides(tools.load_descriptor(overrides_path))
```
Comment (goldmann) [inline concreate/generator.py]: Same here, I would not call it descriptor anymore.

File: concreate/module.py [hunk 3]
```diff
@@ -95,7 +96,7 @@ class Module(Descriptor):
     descriptor_path: A path to module descriptor file.
     """
     def __init__(self, descriptor_path):
-        self.descriptor = Descriptor(descriptor_path, 'module').process()
+        self.descriptor = Module(tools.load_descriptor(descriptor_path))
```
Comment (goldmann) [inline concreate/module.py]: And here too.

File: concreate/generator.py (hunk 1 above)
Comment (dbecvarik) [reply concreate/generator.py]: Do you have any suggestion? Its still a descriptor - only difference is that it is divided into containers not plain dictionaries.

**Comment by dbecvarik** [comment]
@goldmann can you please review again?

**Pull request merged**